pyam/core.py

```diff
@@ -682,15 +681,14 @@ def validate(self, criteria={}, exclude_on_fail=False):
         exclude_on_fail : bool, optional
             flag scenarios failing validation as `exclude: True`
         """
-        df = _apply_criteria(self.data, criteria, in_range=False)
+        df = _apply_criteria(self._data, criteria, in_range=False)
 
         if not df.empty:
             msg = '{} of {} data points do not satisfy the criteria'
             logger.info(msg.format(len(df), len(self.data)))
 
             if exclude_on_fail and len(df) > 0:
-                self._exclude_on_fail(df)
-
+                self._exclude_on_fail(df.to_frame().reset_index())
```

Comment:
it shouldn't ne necessary to do `df.to_frame().reset_index()` if you replace `_meta_idx` by `_make_idx` in the `_exclude_on_fail` function